homeassistant/components/zha/core/channels/general.py

```diff
@@ -89,11 +89,12 @@ async def async_configure(self):
 
     async def async_initialize(self, from_cache):
         """Initialize channel."""
-        power_source = await self.get_attribute_value(
-            "power_source", from_cache=from_cache
-        )
-        if power_source is not None:
-            self._power_source = power_source
+        if not self._ch_pool.skip_configuration or from_cache:
```

Comment:
I feel like we may want to add tests to ensure this behaves exactly as you intend.
For none Xiaomi devices uncached reads will skip this branch entirely. Is this the intended behavior?

Reply:
Non-xiaomi will have skip_configuration = False, so it will enter the if block

Reply:
I can take a look at adding tests still.

Reply:
> non Xiaomi devices uncached reads
as in  `skip_configuration = False` and `from_cache=False`

Reply:
nvm, I get it now